apis/python/src/tiledbsoma/util.py

```diff
@@ -191,3 +191,50 @@ def dense_index_to_shape(
         return stop - start + 1
 
     raise TypeError("coordinates must be tuple of int or slice")
+
+
+def check_type(
+    name: str,
+    actual_type: Type[Any],
```

Comment:
@thetorpedodog and I prefer to pass type of thing. I know most everything should be pass by reference, but, I don't want any chance of overhead of value-copies. I prefer it this way.

Reply:
Everything in Python is a pointer; no matter what you pass in a function call—int, bytearray, table—it’s always passing a reference to some object.

Reply:
@thetorpedodog change of heart, I give in to help smooth this along.

Also, if high-memory-footprint things were being passed by value someaccidentallyhow, it would have been causing issues elsewhere in the callstack already.